packages/dds/sequence/src/test/sharedString.spec.ts

```diff
@@ -236,6 +236,36 @@ describe("SharedString", () => {
 			assert.equal(simpleMarker.properties?.color, "blue", "Could not annotate marker");
 		});
 
+		it("fails when the marker is updated", () => {
+			sharedString.insertText(0, "hello world");
+			// Insert a simple marker.
+			sharedString.insertMarker(6, ReferenceType.Simple, {
+				[reservedMarkerIdKey]: "markerId",
+			});
+			// Annotate the marker.
+			const props = { color: "blue" };
+			const simpleMarker = sharedString.getMarkerFromId("markerId") as Marker;
+			sharedString.annotateMarker(simpleMarker, props);
+			assert.equal(simpleMarker.properties?.color, "blue", "Could not annotate marker");
+			// Annotate the marker's ID.
+			const newIdProps = { [reservedMarkerIdKey]: "newIdValue" };
+			try {
+				sharedString.annotateMarker(simpleMarker, newIdProps);
+				assert.fail();
+			} catch (error) {
+				assert.notEqual(
+					error,
+					undefined,
+					"Error from attempting to update marker was not thrown",
+				);
```

Comment:
Can this ever happen inside a catch block? I'd expect `error` to always be defined here. However, the "error not thrown" message seems like a good fit for the `assert.fail()` in line 254.